harness/determined/common/experimental/checkpoint/_checkpoint.py

```diff
@@ -16,23 +17,40 @@ class ModelFramework(enum.Enum):
 
 
 class CheckpointState(enum.Enum):
-    UNSPECIFIED = 0
-    ACTIVE = 1
-    COMPLETED = 2
-    ERROR = 3
-    DELETED = 4
+    UNSPECIFIED = bindings.determinedcheckpointv1State.STATE_UNSPECIFIED.value
+    ACTIVE = bindings.determinedcheckpointv1State.STATE_ACTIVE.value
+    COMPLETED = bindings.determinedcheckpointv1State.STATE_COMPLETED.value
+    ERROR = bindings.determinedcheckpointv1State.STATE_ERROR.value
+    DELETED = bindings.determinedcheckpointv1State.STATE_DELETED.value
 
 
 @dataclasses.dataclass
 class CheckpointTrainingMetadata:
     experiment_config: Dict[str, Any]
-    experiment_id: str
-    trial_id: str
+    experiment_id: int
```

Comment:
How come those were ever `str`?

Reply:
Pretty sure the proto always had them ints, and this type annotation was just wrong (and would have been wrong at runtime).

from_json() didn't let mypy find this error before.